src/pages/Update.jsx

```diff
@@ -29,19 +34,25 @@ const Update = () => {
   }
 
   useEffect(() => {
-    const fetchCocktail = async () => {
-      try {
-        const data = await getOneCocktail(id)
-        setName(data.name)
-        setMethod(data.method)
-        setRating(data.rating)
-      } catch (err) {
-        navigate('/', { replace: true })
+    const setCocktail = async () => {
+      const cocktail = await data[0]
```

Comment:
don't know exactly how queries work on the RTK (😝 ) but this seems very weird. Are you sure data is an array? and that the first element is a promise? Shouldn't `data` already contain the response from `useGetOneCocktailQuery`? Would you care to elaborate?

Reply:
Maybe I did something wrong in the query, it looks like I'm filtering the data and that's why I get an array as response. I'm not sure about how to just select one row and get that returned as an object. 
When I use the Supabase library it happens automatically.